libraries/Matter/src/MatterEndpoints/MatterThermostat.h

```diff
@@ -0,0 +1,255 @@
+    THERMOSTAT_SEQ_OP_COOLING_HEATING_REHEAT = (uint8_t) Thermostat::ControlSequenceOfOperationEnum::kCoolingAndHeatingWithReheat,
+  };
+  
+  enum ThermostatMode_t {
+    THERMOSTAT_MODE_OFF  = (uint8_t) Thermostat::SystemModeEnum::kOff,
+    THERMOSTAT_MODE_AUTO = (uint8_t) Thermostat::SystemModeEnum::kAuto,
+    THERMOSTAT_MODE_COOL = (uint8_t) Thermostat::SystemModeEnum::kCool,
+    THERMOSTAT_MODE_HEAT = (uint8_t) Thermostat::SystemModeEnum::kHeat,
+  };
+
+  enum ThermostatAutoMode_t {
+    THERMOSTAT_AUTO_MODE_DISABLED = (uint8_t) Thermostat::SystemModeEnum::kOff,
+    THERMOSTAT_AUTO_MODE_ENABLED  = (uint8_t) Thermostat::SystemModeEnum::kAuto,
+  };
+  // clang-format on
+
+  MatterThermostat();
+  ~MatterThermostat();
+  // begin Matter Thermostat endpoint with initial Operation Mode
+  bool begin(ControlSequenceOfOperation_t controlSequence = THERMOSTAT_SEQ_OP_COOLING, ThermostatAutoMode_t autoMode = THERMOSTAT_AUTO_MODE_DISABLED);
+  // this will stop processing Thermostat Matter events
+  void end();
+
+  // set the Thermostat Mode
+  bool setMode(ThermostatMode_t mode);
+  // get the Thermostat Mode
+  ThermostatMode_t getMode() {
+    return currentMode;
+  }
+  // returns a friendly string for the Fan Mode
+  static const char *getThermostatModeString(uint8_t mode) {
+    return thermostatModeString[mode];
+  }
+
+
+  // get the Thermostat Control Sequence of Operation
+  ControlSequenceOfOperation_t getControlSequence() {
+    return controlSequence;
+  }
+
+  // get the minimum heating setpoint in 1/100th of a Celsio degree
+  float getMinHeatSetpoint() {
+    return (float)kDefaultMinHeatSetpointLimit / 100.00;
+  }
+  // get the maximum heating setpoint in 1/100th of a Celsio degree
+  float getMaxHeatSetpoint() {
+    return (float)kDefaultMaxHeatSetpointLimit / 100.00;
+  }
+  // get the minimum cooling setpoint in 1/100th of a Celsio degree
+  float getMinCoolSetpoint() {
+    return (float)kDefaultMinCoolSetpointLimit / 100.00;
+  }
+  // get the maximum cooling setpoint in 1/100th of a Celsio degree
+  float getMaxCoolSetpoint() {
+    return (float)kDefaultMaxCoolSetpointLimit / 100.00;
+  }
+  // get the deadband in 1/10th of a Celsio degree
+  float getDeadBand() {
+    return (float)kDefaultDeadBand / 10.00;
+  }
+
+  // generic function for setting the cooling and heating setpoints - checks if the setpoints are valid
+  // it can be used to set both setpoints at the same time or only one of them, by setting the other to (float)0xffff
+  // Heating Setpoint must be lower than Cooling Setpoint
+  // When using AUTO mode the Cooling Setpoint must be higher than Heating Setpoint by at least the 2.5C (deadband)
+  // Thermostat Matter Server will enforce those rules and the Max/Min setpoints limits as in the Matter Specification 
+  bool setCoolingHeatingSetpoints(double _setpointHeatingTemperature, double _setpointCollingTemperature) {
```

Comment:
any reason why you do long methods in the header? How do you decide wether a method should be implemented in the header or in the source?

Reply:
I'll move it to cpp.

Reply:
Done.